src/components/sidebar/Sidebar.tsx

```diff
@@ -16,13 +16,68 @@ import {
 import { CloseIcon, EditIcon, SearchIcon } from '@chakra-ui/icons'
 import DragItem from './DragItem'
 import { menuItems, MenuItem } from '~componentsList'
+import { useSelector } from 'react-redux'
 import {
-  cmenuItems,
-  CMenuItem,
-} from '../../custom-components/customComponentsList'
+  getCustomComponents,
+  getSelectedCustomComponentId,
+} from '~core/selectors/customComponents'
+import useDispatch from '~hooks/useDispatch'
+import API from '~custom-components/api'
 
 const Menu = () => {
   const [searchTerm, setSearchTerm] = useState('')
+  const dispatch = useDispatch()
+  const currentComponents = useSelector(getCustomComponents)
+  const selectedComponent = useSelector(getSelectedCustomComponentId)
+
+  const getObjectDiff = (updatedList: Record<string, unknown>) => {
+    let deletedComponents = Object.keys(currentComponents).filter(
+      component => !Object.keys(updatedList).includes(component),
+    )
+    let newComponents = Object.keys(updatedList).filter(
+      component => !Object.keys(currentComponents).includes(component),
+    )
+    return {
+      deletedComponents: deletedComponents,
+      newComponents: newComponents,
+    }
+  }
+
+  useEffect(() => {
+    const interval = setInterval(async () => {
+      const newComponentsList = await API.get('/refresh').then(res => res.data)
+      const componentDiffs = getObjectDiff(newComponentsList)
+      if (componentDiffs.deletedComponents.length) {
+        componentDiffs.deletedComponents.map(async component => {
+          const response = await API.post('/delete-file', {
+            path: currentComponents[component],
+          })
+        })
+      }
+      if (componentDiffs.newComponents.length) {
+        componentDiffs.newComponents.map(async component => {
+          const response = await API.post('/init', {
+            path: newComponentsList[component],
+          })
+        })
+      }
+      dispatch.customComponents.updateCustomComponents(newComponentsList)
+    }, 3000)
+
+    return () => {
+      clearInterval(interval)
+    }
+  }, [currentComponents])
+
+  const handleEditClick = async (name: string) => {
+    const response = await API.post('/read-json', {
+      path: currentComponents[name],
+    })
+    console.log(response)
+    dispatch.customComponents.select(name)
+    dispatch.components.reset(JSON.parse(response.data.content))
```

Comment:
Why?

Reply:
This is responsible for importing the component.json of the new component we selected. If we don't do this, it will still have the component.json of the former component loaded and will override the new one with the old one.

Reply:
okay